proposals/0245-sharing-wifi-ssid-and-password.md

```diff
@@ -18,627 +18,1203 @@ If WiFi credentials are automatically generated by the IVI system, the password
+    <td>External</td>
+  </tr>
+  <tr>
+    <td rowspan="2">false</td>
+    <td rowspan="2">true</td>
+    <td>Vehicle/Mobile/omitted</td>
+    <td>Vehicle/Mobile/omitted</td>
+    <td>not be initialized</td>
+  </tr>
+  <tr>
+    <td>External</td>
+    <td>Vehicle/Mobile/omitted</td>
+    <td>External</td>
+  </tr>
+  <tr>
+    <td rowspan="2">false</td>
+    <td rowspan="2">false</td>
+    <td>Vehicle/Mobile/omitted</td>
+    <td>Vehicle/Mobile/omitted</td>
+    <td>not be initialized</td>
+  </tr>
+  <tr>
+    <td>External</td>
+    <td>Vehicle/Mobile/omitted</td>
+    <td>External</td>
+  </tr>
+  <tr>
+    <td rowspan="4">false</td>
+    <td rowspan="4">false</td>
+    <td>true</td>
+    <td>true</td>
+    <td rowspan="4">any</td>
+    <td rowspan="4">any</td>
+    <td rowspan="4">not be initialized</td>
+  </tr>
+  <tr>
+    <td>true</td>
+    <td>false</td>
+  </tr>
+  <tr>
+    <td>false</td>
+    <td>true</td>
+  </tr>
+  <tr>
+    <td>false</td>
+    <td>false</td>
+  </tr>
+</table>
+
+
+__Note:__ `preferredNetworkHost` parameter will be considered only if `hostingWiFiSupported` and `autoJoinWiFiSupported` is `true` in vehicle and mobile application capabilities.
+
+In any case, from the above table, SDL Core has to check whether the network host is allowed in the ini file.
+
+If both HMI and mobile application can host the network and both network host values are allowed in the ini file, SDL Core checks `preferredNetworkHost` from capabilities. If this parameter coincides for mobile and HMI capabilities, then this value has to be taken as a network host.
+
+If the `preferredNetworkHost` is omitted in mobile application network capabilities, vehicle preference will be chosen.
+
+If the `preferredNetworkHost` is omitted in vehicle network capabilities, mobile application preference will be chosen.
+
+If for both vehicle and mobile application network capabilities `preferredNetworkHost` parameter is omitted, a decision will be based on ini file configuration.
+
+If `preferredNetworkHost` of mobile application network capabilities contradict the corresponding parameter of vehicle network capabilities, a decision will be based on ini file configuration.
+
+#### INI file update
+
+```ini
+[Networking]
+;networkHostPriority defines the priority of network host in case parameter is omitted for mobile and vehicle networking capabilities, or if they contradict one another.
+networkHostPriority = VEHICLE, MOBILE
+
+;allowAccessPointSharing defines whether SDL Core is allowed to share credentials of the external access point
+allowAccessPointSharing = FALSE
 ```
 
-### HMI API changes
+__Note:__ `preferredNetworkHost` is the array with priority defined by the order of elements.
 
-1. Device Info is changed to provide the apps with WiFi capabilities to the IVI. The IVI will receive `DeviceInfo.wifiConfigurationInfo` in `OnAppRegistered` HMI API when the app is registered using `primary transport`. 
+### Feature initialization

```

Comment:
The solution from #799 is the first step that we agreed on. But there were a lot of open questions that were needed to be resolved. After the workshop with deeper analysis not all issues and race conditions were fixed. So the proposed solution takes #799 as basis, but adds points that are required for getting things to work.

Reply:
The Steering Committee decision was to return for revisions, to update to reflect the solution in [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/799#issuecomment-588434670).  Please update this PR accordingly.  Once the revised proposal is brought back into review, you may comment with any additional changes you would like to see within the proposal, and the Steering Committee can review and vote on those.

Reply:
FIxed according to https://github.com/smartdevicelink/sdl_evolution/issues/799#issuecomment-588434670

Reply:
The charts in this section still need to be updated to follow: Core requests HMI sets up a WiFi network as an `OnSystemCapabilities` notification that includes `NetworkingCapabilities` and `VEHICLE` as the `preferredNetworkHost`.

Reply:
Fixed spelling OnSystemCapabilityUpdated 
Fixed adding NetworkingCapabilities parameter instead of just wifi wifiFrequencyBandsSupported  
![vehicle_access_point](https://user-images.githubusercontent.com/11158563/82305427-477a6780-99c6-11ea-8f0b-1057174e4f91.png)

Reply:
Please note the arrows below.  There is no param `NetworkConfiguration`, there should only be `NetworkCapabilities`

![image (3)](https://user-images.githubusercontent.com/22600496/82496763-39f2d980-9abb-11ea-9588-5c9efedea416.png)

Reply:
Thanks! Replaced  NetworkConfiguration to NetworkCapabilities in all diagrams